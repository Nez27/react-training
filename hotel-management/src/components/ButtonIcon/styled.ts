import styled, { css } from 'styled-components';

interface IStyledButtonIcon {
  isHaveChildren: boolean;
  style?: {
    fontSize?: string;
    backgroundColor?: string;
    color?: string;
  };
  iconStyle?: {
    color?: string;
    size?: string;
  };
}

const StyledButtonIcon = styled.button<IStyledButtonIcon>`
  background: none;

  padding: 10px;
  transition: all 0.2s;

  border: none;

  ${(props) =>
    props.style?.fontSize &&
    css`
      font-size: ${props.style.fontSize};
    `}

  ${(props) =>
    props.style?.fontSize &&
    css`
      font-size: ${props.style.fontSize};
    `}

  ${(props) =>
    props.color &&
    css`
      font-size: ${props.color};
    `}

  ${(props) =>
    props.isHaveChildren
      ? css`
          display: flex;
          align-items: center;
          gap: 10px;

          text-align: left;
          width: 100%;
        `
      : css`
          border-radius: var(--radius-md);
        `}

  &:hover {
    background-color: var(--hover-background-color);
  }

  & svg {
    ${(props) =>
      props.iconStyle?.size &&
      css`
        width: ${props.iconStyle.size};
        height: ${props.iconStyle.size};
      `}

    ${(props) =>
      props.iconStyle?.color &&
      css`
        color: ${props.iconStyle.color};
      `}
    transition: all 0.3s;
  }
`;

StyledButtonIcon.defaultProps = {
  style: {
    fontSize: '14px',
    backgroundColor: 'var(--primary-color)',
    color: '#000',
  },
  iconStyle: {
    color: '#000',
    size: '14px',
  },
};

export { StyledButtonIcon };
