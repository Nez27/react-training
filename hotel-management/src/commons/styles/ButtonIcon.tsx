import styled from 'styled-components';

const ButtonIcon = styled.button`
  background: none;
  border: none;
  padding: 8px;
  transition: all 0.2s;
  border-radius: var(--radius-md);

  &:hover {
    background-color: var(--hover-background-color);
  }

  & svg {
    width: 25px;
    height: 25px;
  }
`;

export default ButtonIcon;
