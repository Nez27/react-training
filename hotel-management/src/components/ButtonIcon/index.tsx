import { ButtonHTMLAttributes, ReactNode } from 'react';

// Styled
import { StyledButtonIcon } from './styled';

interface IButtonIcon extends ButtonHTMLAttributes<HTMLButtonElement> {
  icon: ReactNode;
  iconSize?: string;
  iconColor?: string;
  fontSize?: string;
  text?: string;
  variations?: string;
}

const ButtonIcon = ({
  icon,
  iconSize,
  iconColor,
  text,
  fontSize,
  variations,
  ...props
}: IButtonIcon) => {
  return (
    <StyledButtonIcon
      iconColor={iconColor}
      iconSize={iconSize}
      fontSize={fontSize}
      variations={variations}
      {...props}
    >
      {icon} {text && <span>{text}</span>}
    </StyledButtonIcon>
  );
};

export default ButtonIcon;
