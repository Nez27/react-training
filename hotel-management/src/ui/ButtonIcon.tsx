import styled from 'styled-components';

const ButtonIcon = styled.button`
  background: none;
  border: none;
  padding: 8px;
  transition: all 0.2s;
  border-radius: var(--radius-sm);

  &:hover {
    background-color: #f3f4f6;
  }

  & svg {
    width: 25px;
    height: 25px;
  }
`;

export default ButtonIcon;
