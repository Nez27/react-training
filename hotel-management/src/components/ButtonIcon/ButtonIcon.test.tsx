import { HiOutlineLogout } from 'react-icons/hi';

// Components
import { render } from '@testing-library/react';
import ButtonIcon from '.';

describe('ButtonIcon', () => {
  const wrapper = render(
    <ButtonIcon
      aria-label="Logout"
      icon={<HiOutlineLogout />}
    />
  );

  test('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
