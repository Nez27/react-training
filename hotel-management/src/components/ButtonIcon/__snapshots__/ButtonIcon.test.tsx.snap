// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="Logout"
        class="sc-bdnyFh gPNQLR"
        iconcolor="#000"
        iconsize="16px"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
         
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="Logout"
      class="sc-bdnyFh gPNQLR"
      iconcolor="#000"
      iconsize="16px"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
       
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
